packages/embed/src/factories/create-popup/create-popup.ts

```diff
@@ -10,17 +10,43 @@ export type Popup = {
 
 export const createPopup = (formUrl: string, options: PopupOptions): Popup => {
   const iframe = createIframe(formUrl, 'popup', options)
+  const popup = buildPopup(iframe)
+  const container = options.container || document.body
 
-  const popup = document.createElement('div')
-  popup.className = 'typeform-popup'
-  popup.append(iframe)
+  const open = () => {
+    container.append(popup)
+  }
 
-  const container = options.container || document.body
-  container.append(popup)
+  const close = () => {
+    if (popup.parentNode) {
+      popup.parentNode.removeChild(popup)
+    }
+  }
+
+  popup.append(buildCloseButton(close))
+
+  const refresh = () => {
+    iframe.contentWindow?.location.reload()
+  }
 
   return {
-    open: () => popup.classList.add('typeform-popup-opened'),
-    close: () => popup.classList.remove('typeform-popup-opened'),
-    refresh: () => iframe.contentWindow?.location.reload(),
+    open,
+    close,
+    refresh,
   }
 }
+
+export const buildPopup = (iframe: HTMLIFrameElement): HTMLDivElement => {
```

Comment:
I prefer this here in this file while its short. If it becomes very long we can extract it to separate file. WDYT?

Reply:
On the other hand we would need to do `export { createPopup } from './create-popup'` in `./index.ts` to make sure we dont export this to library consumers. I am open to ideas.

Reply:
My first thought was - why even exporting? - and I see you need it for unit tests.

Well, then we can place these helper functions like this:
```
create-popup/
  utils/
    build-popup.ts
    build-close-button.ts
    index.ts
  create-popup.ts
  create-popup.spec.ts
  index.ts
```
so we both export the functions from their source files for testing, and do not export them outside the create-popup.

Reply:
Good idea. I will name the directory `elements` not `utils`, since anything can be "utils".